packages/javascript/bh-shared-ui/src/components/HelpTexts/CodeController/CodeController.tsx

```diff
@@ -0,0 +1,155 @@
+ * @param hideCopy - default true - display copy text button
+ * @param hideWrap - default true - display wrap text button
+ * @returns
+ */
+function CodeController(props: PropsWithChildren<Props>) {
+    const { hideCopy = false, hideWrap = false, children } = props;
+
+    const [wrapped, setWrapped] = useState(true);
+    const [copied, setCopied] = useState(false);
+    const [scrollLeft, setScrollLeft] = useState(false);
+    const [scrollRight, setScrollRight] = useState(false);
+
+    const codeRef = useRef<HTMLPreElement>(null);
+
+    const classes = useStyles();
+
+    const handleScroll = (e: React.UIEvent<HTMLPreElement>) => {
+        const { scrollLeft, scrollWidth, clientWidth } = e.currentTarget;
+        setScrollLeft(scrollLeft + clientWidth < scrollWidth);
+        setScrollRight(scrollLeft > 0);
+    };
+
+    const handleCopy = async () => {
+        setCopied(true);
+        // Trims off the white space at the beginning and end of new lines
+        const justifiedLeft = (children?.toString() ?? '')
+            .split('\n')
+            .map((s) => s.trim())
+            .join('\n');
+
+        await copyToClipboard(justifiedLeft);
+
+        setTimeout(() => {
+            setCopied(false);
+        }, 3000);
+    };
+
+    const handleWrap = () => {
+        setWrapped((prev) => {
+            if (prev) setScrollLeft(true);
+            return !prev;
+        });
+    };
+
+    return (
+        <div className={classes.codeController}>
+            <Typography
+                component='pre'
+                className={clsx('code', {
+                    wrapped,
+                    scrollLeft: !wrapped && scrollLeft,
+                    scrollRight: !wrapped && scrollRight,
+                })}
+                ref={codeRef}
+                onScroll={handleScroll}>
+                {(!hideCopy || !hideWrap) && (
+                    <>
+                        <div className='codeController'>
+                            {!hideCopy && (
+                                <Button sx={{ p: 0.5, m: 0, fontSize: '12px' }} onClick={handleCopy}>
+                                    <FontAwesomeIcon icon={faCopy} />
+                                    <Typography component='span' sx={{ marginLeft: '6px' }}>
+                                        {copied ? 'Copied' : 'Copy'}
+                                    </Typography>
+                                </Button>
+                            )}
+                            {!hideWrap && (
+                                <Button
+                                    sx={{ p: 0.5, m: 0, marginRight: '20px', fontSize: '12px' }}
+                                    onClick={handleWrap}>
+                                    <FontAwesomeIcon icon={faAlignJustify} />
+                                    <Typography component='span' sx={{ marginLeft: '6px' }}>
+                                        {wrapped ? 'Unwrap' : 'Wrap'}
```

Comment:
I see that the wrap toggle for the state is happening when I click the button but the text does not look to be wrapping. This may have been implemented and tested in BHE where the entity panel styling is different so the wrapping may not be happening the same way.

Reply:
The firefox checks out! Good find and fix